test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -1800,6 +1800,43 @@ TEST_F(LuaHttpFilterTest, GetRequestedServerName) {
   EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, true));
 }
 
+// Verify that extracting dynamicMetadata() using LUA also works for binary values.
+TEST_F(LuaHttpFilterTest, DynamicMetadataBinaryData) {
+  const std::string SCRIPT{R"EOF(
+    function envoy_on_request(request_handle)
+      local bin_data = request_handle:streamInfo():dynamicMetadata():get("envoy.pp")["bin_data"]
+      local hex_table = { }
+      local data_length = 8
+      for idx = 1, data_length do
+        local hex_value = string.format('%x', string.byte(bin_data, idx))
+        if string.len(string.format('%x', string.byte(bin_data, idx))) < 2 then
+          hex_table[data_length - idx + 1] = '0'..hex_value
+        else
+          hex_table[data_length - idx + 1] = hex_value
+        end
+      end
+      local hex_data = table.concat(hex_table, '')
+      local int64_data = string.format('%d', tonumber(hex_data, 16))
```

Comment:
This feels unrelated to the functionality. While yes, you could encode a number with this data, we don't really need to test that part. Why not just stick some binary data in there, and make sure to include some \0 since that's what pushstring chokes on otherwise, and print those as escaped hex (or whatever other way printing is easy) to verify.